### PR TITLE
Adding Deposit Modal in profile

### PR DESCRIPTION
**Pull request by Alejoboga20**
**Pull request type**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Migration
- [ ] Other (please describe):

**What changes does this PR introduce?**

- Adding Deposit Section in My Balances
- Creating modal to display default credit card with form for deposit.

Issue Link: https://rootstrap.atlassian.net/jira/software/projects/DOL/boards/93?selectedIssue=DOL-209

**Security Considerations**

- [ ] Perform server-side input validation and sanitization (whenever it's possible) on all user-controlled inputs.
- [ ] Perform context-aware output encoding/escaping of all user-controlled input.
- [ ] Enforce Access Control on every request. Consider the different types of users. Who should access that functionality/endpoint?
- [ ] Requires Test updates.

**Other information**

N/A

**Screenshots**
**Pull request type**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Migration
- [ ] Other (please describe):

**What changes does this PR introduce?**

- Refactoring Home Page.
- Creating constants with home carousel items.
- Adding control to make transition automatic
- Implement Automatic Carousel with banners for Home page

Issue Link: https://rootstrap.atlassian.net/jira/software/projects/DOL/boards/93?selectedIssue=DOL-206

**Security Considerations**

- [ ] Perform server-side input validation and sanitization (whenever it's possible) on all user-controlled inputs.
- [ ] Perform context-aware output encoding/escaping of all user-controlled input.
- [ ] Enforce Access Control on every request. Consider the different types of users. Who should access that functionality/endpoint?
- [ ] Requires Test updates.

**Other information**

N/A

**Screenshots**
image
image

### CODE REVIEW

File: src/app/pages/Profile/UserBalance/UserBalance.tsx [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 import { Convert } from 'app/components/Convert/Convert';
+import { FundWalletModal } from 'app/components/FundWalletModal/FundWalletModal';
 import useTranslation from 'app/hooks/useTranslation';
 import styles from './UserBalance.module.scss';
 
 export const UserBalance = () => {
   const t = useTranslation();
   const { balances } = useUserBalance();
   const { isMobileView } = useResponsive();
+  const [isFundtModalOpen, setIsFundModalOpen] = useState(false);
```
Comment (gastonri) [inline src/app/pages/Profile/UserBalance/UserBalance.tsx:14]: Typo on "Fundt".

File: src/app/components/FundWalletModal/useFundWallet.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    });
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setDeposit({ ...deposit, [name]: value });
+  };
+
+  useEffect(() => {
+    return () => {
```
Comment (gastonri) [inline src/app/components/FundWalletModal/useFundWallet.ts:38]: Why not adding this return on the other useEffect? Either way, the component is going to be unmounted, and it has nothing responding to open param.

File: src/app/locales/en.json [hunk 3]
```diff
@@ -101,8 +101,17 @@
...
       "availableBalanceColumn": "Available Balance",
+      "depositButton:": "Make a Deposit",
       "usdValueColumn": "USD Value"
     },
+    "deposit": {
+      "title": "Deposit currency via credit card",
+      "description": "FTX US uses Circle to allow you to safely and securely deposit using your credit or debit card (issued by MasterCard/VISA only).",
+      "info": "You may deposit any currency, and it will be automatically converted to USD.",
+      "smallCvv": "This is used for card verification Amount",
+      "smallAmount": "You will be charged an additional $0.31 in fees. These fees come from our external processor.",
```
Comment (gastonri) [inline src/app/locales/en.json:112]: It's always a fixed fee?

**Review by gastonri — COMMENTED** [review]
Looks awesome! 💪

File: src/app/locales/en.json (hunk 3 above)
Comment (Alejoboga20) [reply src/app/locales/en.json:112]: Yes, that's a fixed value for deposits . Took the text from ftx.

File: src/app/components/FundWalletModal/useFundWallet.ts (hunk 2 above)
Comment (Alejoboga20) [reply src/app/components/FundWalletModal/useFundWallet.ts:38]: I will move this to only one effect, thanks

**Review by gastonri — APPROVED** [review]

**Pull request merged**